### ISSUE TITLE
No-Colour Icons

### ISSUE THREAD

**Issue by redbassett** [CellularPrivacy/Android-IMSI-Catcher-Detector#512]
Somewhat relevant to #479, but in the other direction: With the shift toward monochrome notifications icons, I would love to see an optional set of icons that could display all of the relevant states, but with a single colour to match. Currently, this is the one of only two persistent notification icon apps that use colour, and it doesn't match the rest of my notifications bar. Having this as a option UI enhancement for users would be a minor task but great improvement.

**Comment by benutzer74** [comment]
I fully agree!
I DAMN HATE this whiteish Lollipop cr@p.
We definetely do need colored icons to easily recognize important messages.
AIMSICD definetely need colored status bar icons to easily show current state of alert.
Google has to understand, the flat white colors in Lollipop is not only a mess, it is a danger because you may miss important notifications.

**Comment by DJaeger** [comment]
I have already posted this in #479.
But we can move this enhancement into this issue.

**Comment by redbassett** [comment]
To clarify, opposite of @benutzer74, I want the all-white icons, but I want to still see the app's status. (I also hav no idea how the current icons are for colourblind users?)

I would like an icon set that can be presented in one colour and still show this status.

@DJaeger: This is connected to your issue (hence why I mentioned it up top), however I see it as a different development task since yours involves overriding Lollipop when it auto-changes the icon, and I'm asking for a development of a new user preference and a new graphics set.

**Comment by DJaeger** [comment]
I know what you mean, as this is what I also suggested:
>You should add "real" white icons out of the "white_" icons by making the transparent parts inside white and removing all other outsite.

>Before looking why it differs from time to time the icon should be set as the doku says, that we don't search for an issue caused by bad behavier of Lillipop to make the icons white.

>On making icons white they have to be changed, that the state could be determined without color

**Comment by redbassett** [comment]
Somehow after several read-overs, I completely missed that bit. Absolutely!

**Comment by SecUpwN** [comment]
>Somehow after several read-overs, I completely missed that bit. Absolutely!

@redbassett, thanks for opening this Issue and developing with us! Here is my take on the whole thing: Google will *very likely* realize that they've moved down a shitty road with the whitish notifications and *hopefully* change that in upcoming versions of Android. Since this Issue you've opened here involves a whole new re-design of another Icon set, more preferences and likely a huge chunk of code just to tackle this, I'd vote for **closing** this Issue in favor of #479. What do you think, @He3556 and @E3V3A?

**Comment by E3V3A** [comment]
Agree

**Comment by DJaeger** [comment]
@SecUpwN: Why should Google go back?
Most apps, except this one, use white icons as the doku says!
Why does this involve a redesign of other icons and code?

**Comment by SecUpwN** [comment]
>Most apps, except this one, use white icons as the doku says!

Yes, and that is what I hate about Google: Once they change Android, they also just change their documentation telling all developers "Hey dudes, here is what every App should now do. Go ahead and work your asses off to change it!" - and when a company does that, I like using my own brain.

>Why does this involve a redesign of other icons and code?

Strange question. Maybe because we'd actually have to craft new Icons in black/white and shapes? As already said, the best solution for all this trouble would be fixing #479, which will make our app display the Status-Icons as they should be displayed. This whitish style in the **notification** bar really nukes the whole sense of *notification* all by itself... and it#s freaking ticking me off!

**Comment by E3V3A** [comment]
:+1:

**Comment by redbassett** [comment]
Glad to see some debate on the issue! I personally like the white notifications. I'm fine with this being closed as an issue for primary development, but maybe some community members would be able to contribute to the project by designing something like this? I am not one for cluttered settings panels, but I would love to have the choice to match this app to the other persistent notification apps I have.

**Comment by DJaeger** [comment]
The main purpose of the status bar is to tell the user: "There is something to mention!" and if you want how much to mention (with count shown on your icon).
It is not to tell the user "what to mention".
For that there are the notifications, where you can show what ever you want with how much informations you want.
So for this purpose it does not matter if icon are colorful or not, they only have to tell you "app xyz has something for you".

**Comment by E3V3A** [comment]
@DJaeger I'm glad that you're so aligned with Googles Android vision, which is why we love having you on our team. 

But there is no reason at all why in this fucking day and age of zillion colored UHD screens that we should not be able to see color if we want to. How about starting a political party where we change all the street stop-lights to Googles version? After all its easy to do, just tell all stop light manufacturers to replace the (now) colored LEDs with white ones, and don't listen to anyone else. That's exactly what the headless dick-wipe Google AOS upper managers are doing.

**Comment by redbassett** [comment]
I understand not liking their design choices and ever-changing standards, but I'm inclined to not be a fan of @E3V3A's expression of this. I like the new style, but I can see why it is a low priority for a project like this. I also am inclined to agree with @DJaeger /re the idea that notifications might be considered events rather than persistent statuses, but that is a major change of philosophy from how this app currently works.

Either way, I was kinda ok with this issue being closed, but seeing the post above mine, I'm starting to think that maybe it wasn't closed for a reason I agree with…

I would like to remind @E3V3A that I proposed the idea of some sort of way for the user to choose coloured icons or white ones, not forcing anyone to abandon the colour if they like it… (And this issue has no bearing on fixing #479, which should be fixed regardless of this enhancement.)

**Comment by E3V3A** [comment]
> I proposed the idea of some sort of way for the user to choose coloured icons or white ones

@redbassett  Thanks for clarifying your intents. Reopening issue.

**Comment by SecUpwN** [comment]
Ok guys, I will get in touch with our graphic designer @SgtObst this weekend then. Stay patient.

**Comment by redbassett** [comment]
Sounds good, thanks!

Apologies for the confusing issue :wink:

**Comment by E3V3A** [comment]
I'm not sure what to make out of [THIS](https://stackoverflow.com/a/29432203/1147688) comment. Is that a solution? Can we use that?

**Comment by SecUpwN** [comment]
>I'm not sure what to make out of THIS comment. Is that a solution? Can we use that?

@E3V3A, what you have posted has already been sent to me by @He3556 and is meant for #479. this Issue is a different one: @redbassett is requesting black/white Icons with different *shapes*. Implementing this would also be another solution to the already solved #449.

**Comment by E3V3A** [comment]
Great!  I was getting confused

**Comment by E3V3A** [comment]
@SecUpwN 

> SgtObst already works on this.

Any progress or just talking? ( I don't mind either, but it's been more than 2 weeks and I'm only asking...)

**Comment by DJaeger** [comment]
@SecUpwN: Is @SgtObst working on the icons itself or the implementation or both?

**Comment by SecUpwN** [comment]
>Any progress or just talking?

@E3V3A, please don't push me, I have a private life to manage as well! Thank you for caring.

>I don't mind either, but it's been more than 2 weeks and I'm only asking...

I am still waiting for the Icons and will then present them here first. Please **be patient**!

>@SgtObst working on the icons itself or the implementation or both?

@DJaeger, likely just the Icons since he seems overloaded with many projects and his exams in the meantime. If you'd like to, you may tackle the implementation once we've agreed on the upcoming Icons.

**Comment by SgtObst** [comment]
@all I am back from sportsbusiness. I will give it a try this night! Stay tuned!

**Comment by redbassett** [comment]
Peanut gallery comments:

:+1: for Sample 1. If Sample 2 was used, I'd recommend including the negative space in the tower icon from Sample 1, but I prefer the loading-bar style with a single icon more than the gradual icon pieces.

**Comment by DJaeger** [comment]
Ok, I will wait for and look into.


-- 

Mit freundlichen Grüßen

Daniel Jäger

**Comment by SecUpwN** [comment]
Thanks for your work, @SgtObst! I am, however, not quite convinced if users can tell the difference between each status when the Icon appears to have sop small changes. Personally, I thought about this:

* Pure black and white Icons
* Different Icon for each status (e.g.: Idle =  clock, Normal = checkmark, Medium = Egg, High = Attention sign, Dangerous = exploding, Run = keep skull) - you may be creative here, but it shall the cool style

Again, these are just my thoughts. What is your opinion here, @He3556 and @E3V3A?

**Comment by SgtObst** [comment]
In my eyes a combination of both would be most matching!

**Comment by DJaeger** [comment]
I would also recommend different icons for different states.
With your samples it is not really clear which one is which state.
The icons should be mostly self-explaining.

**Comment by DJaeger** [comment]
But it should also be clear, that they are from AIMSICD.

**Comment by E3V3A** [comment]
Uhm, I don't think it's a good idea with completely different icons for each state. How would the user know it's from our App and not something else? That would be very confusing. I think we're on to something with sample 2, but I'm not convinced over the antenna icon there. 

I mean perhaps show something stackable like the 1-4 lightning bolts, but they'd need an outline so that we know it's from our app...

**Comment by DJaeger** [comment]
I think example 2 is very confusing as it looks like it shows the radio connectivity.
I think something with the antenna and a icon for each state in front of it.
For example like the counter of the SMS app (at least the one of CM).

**Comment by SecUpwN** [comment]
>I think something with the antenna and a icon for each state in front of it.
>For example like the counter of the SMS app (at least the one of CM).

Yes, I like that idea. Simple, yet obvious that it is from AIMSICD. Any ideas, @SgtObst?

**Comment by SgtObst** [comment]
@DJaeger we can't do. I have only 1 color basically. And guys, pls remember, the icons are 5 millimeter small. There is not much a

**Comment by SgtObst** [comment]
@DJaeger we can't do. I have only 1 color basically. And guys, pls remember, the icons are 5 millimeter small. There is not much to do.

**Comment by E3V3A** [comment]
How about an antenna with a number from 0-3 and then the skull. Or a skull with numbers 0-4 in one of its eyes?

**Comment by DJaeger** [comment]
@SgtObst: I know, where's the problem?
The message icon is also 2 colow and we only have to use an antenna where it is using the message symbol!?
![tmp_32535-_20150803_14005979277631](https://cloud.githubusercontent.com/assets/3408882/9037141/6d85fd60-39e8-11e5-9e8a-9240026a30da.JPG)

**Comment by SecUpwN** [comment]
>I agree.How about a car (!) with a number from 0-3 next to it and then the skull.

I actually like that idea, but I'm not sure if that fits our general app Icon style...

**Comment by DJaeger** [comment]
A car?
Please not.
That is useless to identify it as icon of AISICD...

**Comment by E3V3A** [comment]
:) Yes, bad idea.

**Comment by DJaeger** [comment]
I think an antenna like in the first example would be great in the background of a badge like in my posted image

**Comment by E3V3A** [comment]
yes

**Comment by SecUpwN** [comment]
>I think an antenna like in the first example would be great in the background of a badge like in my posted image

Agreed. So what we have now decided is to use the antenna from our Icons and adding a round circle with the "status number" for each [Status Icon](https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/wiki/Status-Icons) in it? If so, here is what the summary:

* 0 = IDLE
* 1 = NORMAL
* 2 = MEDIUM
* 3 = HIGH
* 4 = DANGER
* Skull = RUN

I am notifying @SgtObst to turn into creative mood as I type this. Patience, please.

**Comment by DJaeger** [comment]
Why not using a very simple and also in this size identifiable icon for the status icons?
Like:
 * IDLE = hourglass
 * NORMAL = no circle / checkmark
 * MEDIUM = warning triangle / i(nfo) sign?
 * HIGH = don't know / warning triangle?
 * DANGER = error icon
 * RUN = Skull

**Comment by E3V3A** [comment]
I prefer number + same icon, over different icons, which are very confusing...

**Comment by DJaeger** [comment]
@E3V3A: Why is it confusing?
The icons are much more user friendly, since the user doesn't have to remember the numbers!?

**Comment by E3V3A** [comment]
Because:  It doesn't make sense as you describe, as hourglass means something is working, and its not, its idling, warning triangle in 2 versions? and what is an error icon? What is the one thing that will ensure the user knows it is using AIMSICD and not something else? 5 icons = bad idea.

**Comment by DJaeger** [comment]
Hourglass means for me "wait" and this is IDLE for, or not?

For MEDIUM and HIGH I mean either the icons before the slash or after it.

Some example icon to visualize them:
https://www.google.com/design/icons/index.html#ic_hourglass_empty
https://www.google.com/design/icons/index.html#ic_check
https://www.google.com/design/icons/index.html#ic_warning
https://www.google.com/design/icons/index.html#ic_error_outline

Why should the user not know, if they all are in front of an antenna???

**Comment by redbassett** [comment]
Icons would probably make sense if used as a badge in front of the app's default (antenna?) icon like was mentioned for numbers.

**Comment by E3V3A** [comment]
How about a modified version of [Track Changes](https://www.google.com/design/icons/index.html#ic_track_changes), with numbers?

**Comment by DJaeger** [comment]
@E3V3A: You mean the Track Changes Icon as new app icon?
Why numbers?

**Comment by TPS** [comment]
Keeping in mind

> @SgtObst commented 4 days ago
> 
> @DJaeger we can't do. I have only 1 color basically. And guys, pls remember, the icons are 5 millimeter small. There is not much to do.

I also vote for **not** using numbers, as most end-users can't keep in mind what they mean, *unless* they actually measure something intuitive (e.g., SMS, reception bars, e-mail, &c).  But I also agree they need a general theme that ties them back to AICD. 

I think our major issue here is that we lack 1 custom b&w icon that belongs specifically to AICD.  Before, when we relied on color, that logo worked well to identify the app, but converting to b&w made that too generic.

Maybe we should rethink the app logo, & then work on overlay variations to *that,* because I honestly don't foresee any other way to satisfy both of these (necessary, IMHO) conditions.  I'm thinking maybe co-opting 1 of innumerable emoji/icons in the Unicode standard (in lieu of my own lack of creativity ;) would be a great start to this idea, as they're **designed** to be legible at tiny sizes, & they readily come in both color & b&w versions.

What do you all think?  I'm planning on listing likely candidates as I find them.

**Comment by TPS** [comment]
After looking through several thousand Unicode characters, a variation of Mercury looks to be a most promising candidate:

![Mercury](https://upload.wikimedia.org/wikipedia/commons/2/2e/Mercury_symbol.svg)

but with variations that I imagine to be (both):

1) Replace the curved horns &/or the body with something that's reminiscent of an antenna — maybe an old style TV aerial?  Perhaps an inverted squat version of our current logo?  With increasing / decreasing # of prongs to show severity?  (To satisfy @E3V3A, &c, who like the # idea. ;)

2) Instead of a blank O for a face, maybe use smilies :smiley: :smiling_imp:  / neutral :expressionless: / frownies :frowning: :imp: / skull :alien: :hankey: :japanese_ogre: :skull: :speak_no_evil:  to further reinforce the severity of warning.  (To satisfy @DJaeger, &c, who like the no-# idea ;)

Comments?

**Comment by E3V3A** [comment]
> vote for not using numbers, as most end-users can't keep in mind what they mean, unless they actually measure something intuitive...

Two counter arguments. Remembering the number scheme is just as easy (if not easier) as remembering the color scheme. They do measure something intuitive, the threat level. 

I'd really hate to see all those emoticon-like icons. It would make our app look childish and unprofessional.

**Comment by DJaeger** [comment]
A "thread level" is not intuitive to "normal" endusers.
But colors are intuitive as they are common to visualize priorizations and how critical something is.
Colors in general have a meaning to users.
The same for smileys.
They are self speaking to visualize states.
But I would use simplified smileys.

**Comment by DJaeger** [comment]
@TPS: I don't see why we should use the Mercury icon!?
I think an antenna is great.

**Comment by TPS** [comment]
@E3V3A As @DJaeger mentions, a threat level isn't something your typical person (outside/aside from government ;) numerically counts.  Maybe a grade: A for excellent, F for failing, X for imminent danger?  The smilies work in the same way, of course, but are, admittedly, more cartoonish.

@DJaeger As I mentioned, the antenna, by itself in b&w, doesn't precisely scream "AICD!" & it doesn't lend itself too well to a security status, without being either overlayed with or incorporated into other symbolism.

@All
For the record, I'm NOT *married* to the symbolism I've suggested!  I just want to think outside the box, instead of sitting on 1 side or the other, with good points on both sides, but, individually, insufficient for the reasons I've mentioned.  Feel free to suggest something else!  Also, I'd like more participants to this end of the discussion (I know, common theme all over this project ;), & I'd especially like to hear from @SecUpwN as repository maintainer, who tends not be terribly partisan without good reason. ☺

**Comment by SecUpwN** [comment]
>@SecUpwN as repository maintainer, who tends not be terribly partisan without good reason. ☺

Ok, this Issue seems to slightly get out of hand. Please stay calm everyone.

>Maybe we should rethink the app logo, & then work on overlay variations to that, because I honestly don't foresee any other way to satisfy both of these (necessary, IMHO) conditions.

Thanks for your great input @TPS, but I don't think that redesigning the whole app Icon is a good option. I also agree with @E3V3A that using smileys in our app does not appear very professional (yes, we're working on growing up), but I do in fact understand the good point you're trying to make.

The core difference we currently have to solve in this Issue is the following:

>I prefer number + same icon, over different icons, which are very confusing...

versus

>The icons are much more user friendly, since the user doesn't have to remember the numbers!?

We first need to solve this difference before we can move on. I don't like the "Google Guidelines", but what do those say in such cases? And are there any sample apps working with a similar case?

**Comment by DJaeger** [comment]
All apps I know shown no state (except notification available or not), an on/off state or a count.
Only exception is Osmand, which shows on running navigation **only** an icon representing the moving type (car/bicycle/walking).

**Comment by TPS** [comment]
@SecUpwN Well, just picking Doom faces (& guessing @ precise app states or equivalents ;) :

:godmode: = A  :suspect: = B  :rage1:  = C  :hurtrealbad: = D  :feelsgood: = F  :finnadie: = X

But going B&W will be a problem again, as these are too detailed; also, how does 1 signify AICD?

**Comment by SecUpwN** [comment]
@TPS, please forget my random thought with DOOM faces. I guess you did not understand why a black/white Icon is a good idea: We'd not only take care of ensuring users are still happy with our app when Google decides to make another set of heavy changes to AOS (so that our workaround [here](https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/blob/development/app/build.gradle#L46) does not work any longer), but we'd also tackle the already closed #449 for good. Makes sense to you?

**Comment by TPS** [comment]
@SecUpwN Oh, I understand: Google screwed us (& folks like us) completely, & now we're trying to hack/kludge around it, though fixing #449 in the process is an excellent idea. ☺  I'm just not finding a good way forward without backing all the way to rethinking the *base* logo/icon.

Do you all know problems like this are so hard that there're entire businesses around this kind of thing?:

http://www.logomyway.com/

**Comment by TPS** [comment]
Just as a note to all who may be still following:  _This_ is what AIMSICD is fighting with in the notification bar:
![Standard Notification Icons 1](http://pwk.republicwireless.com/wp-content/uploads/2015/10/material-action-icons.png)
![Standard Notification Icons 2](http://pwk.republicwireless.com/wp-content/uploads/2015/10/material-image-icons.png)
![Standard Notification Icons 3](http://pwk.republicwireless.com/wp-content/uploads/2015/10/material-device-icons.png)

& this is just standard Android icons, not even other apps! Especially notice that 3rd set of "device" icons.…  How does AIMSICD signify in _that_ lineup?!

(Thanks to Phandroid via RW @ http://pwk.republicwireless.com/android-notification-icons/)

**Comment by SecUpwN** [comment]
I would like to hear what @SgtObst has to say about all this since he designs our Icons. Thanks.

**Comment by larsgrefer** [comment]
I think we should'nt keep the colored statusbar icon, because it forces us to keep our `targetSdk` below 21 (In my opinion a `targetSdk` above 21 is more important than an colored icon)

In order to inform the user about the current status i see 2 options:
- Using differet shapes for different status like here: https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/issues/512#issuecomment-125046784
- Using "peeking notifications" to inform about status changes:
https://www.google.com/design/spec/patterns/notifications.html#notifications-peeking-notifications
  - This peeking notification could provide fast access to actions like "enable airplanemode"
- A combination of both is possible, of course

**Comment by TPS** [comment]
Please don't _force_ those of us who don't _want_ monochromatic notifications into having them!  Will `targetSdk` < 21 prevent installation or functionality on any systems?

**Comment by larsgrefer** [comment]
@TPS What is your opinion about a peeking notification? It should "scream 'AIMSICD'" pretty well.

The decision to use monocromatic statusbar icons is not mine, but the decision of the AOSP when they introduced the Material-Design with Android 5.0.

I don't want to enforce monocromatic icons for this app, but i'd like to use a targetSdk version as high as possible because i want to use native material-design features on devices with api 21 and higher

**Comment by TPS** [comment]
Peeking notifications are great as an _addition_, but not _instead_ of ongoing state notifications in the notification bar.  They work best as change-of-state (i.e., to a worse security cell tower, &c) notices.

Many apps seem to have (configurable) Material skins.  What other features did you anticipate would be important enough to lose the compatibility ___&___ the color notifications for those who want them?

**Comment by larsgrefer** [comment]
> lose the compatibility
What kind of compatibility do you mean?

**Comment by larsgrefer** [comment]
> (In my opinion a `targetSdk` above 21 is more important than an colored icon)

@SecUpwN What's your opinion about this?

**Comment by TPS** [comment]
I guess I worded that incorrectly:  I meant loss of _functionality_ for those devices < targetSdk, with the app relying on features not present in environment.

**Comment by larsgrefer** [comment]
@TPS The App can only rely on features present in `minSdk`. All features above that can only be used conditionally (because they are maybe not present)

The `targetSdk` means:
>This attribute informs the system that you have tested against the target version and the system should not enable any compatibility behaviors to maintain your app's forward-compatibility with the target version. The application is still able to run on older versions (down to `minSdkVersion`).

>As Android evolves with each new version, some behaviors and even appearances might change. However, if the API level of the platform is higher than the version declared by your app's `targetSdkVersion`, the system may enable compatibility behaviors to ensure that your app continues to work the way you expect. You can disable such compatibility behaviors by specifying `targetSdkVersion` to match the API level of the platform on which it's running.

Such a *compatibility behavior* is the reason why we can currently use colored statusbar-icons on api >= 21

**Comment by TPS** [comment]
>The App can only rely on features present in `minSdk`. All features above that can only be used conditionally (because they are maybe not present)

@larsgrefer I don't mind the _additional, optional_ functions, then, as such, but then the devs will have to keep in mind UI dichotomy for the have-nots.  Eventually, the temptation is to raise `minSdk` to match `targetSdkVersion`, as it's too hard to (1) maintain simply optional features or (2) deal with trying to backport features that were conceived as optional, &, with later iteration, become core.

**Comment by TPS** [comment]
Also, some folks don't like peeking notifications (too much like popup ads), so configurable 'off', please!

Instead of increasing `targetSdk`, perhaps importing code from others e.g., https://github.com/SimenCodes/heads-up (GPL3), would do the job?

**Comment by SecUpwN** [comment]
> (In my opinion a `targetSdk` above 21 is more important than an colored icon)
>@SecUpwN What's your opinion about this?

As you already saw, I merged the first part of material design after I've read the postive https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/pull/654#issuecomment-162320271. Regarding future versions of our app, we always had the mindset to make AIMSICD work as best as possible for as many devices as possible. Not sure if that is possible when reasing the `targetSdk`. Can you reference any Android projects on GitHub with the same Issue we currently have here for a decision?

**Comment by larsgrefer** [comment]
@TPS The option to turn of peeking notifications is already provided by the android system

@SecUpwN I agree with you that we should be careful with too much circumventing

As far as I know its a best practice to keep
- `targetSdkVersion` and `compileSdkVersion` as high as possible
  - in order to be able to use newer features on newer phones
- `minSdkVersion` as low as possible
  - in order to be able to run on as many devices as possible
  - @SecUpwN this value decides our compability for older androids

I only know projects which try to follow these *rules*

**Comment by smarek** [comment]
Also, it's probably said somewhere in this thread, but notification on locked screen in lollipop is the same for all states of AIMSICD (disregarding red/yellow/green/...), this should be fixable by the same process

**Comment by DJaeger** [comment]
Again I fully agree with @larsgrefer.

One new feature that is relevant to us introduced in a newer SDK version is the native support of Dual-SIMs and also the material design, which will both be complicated to introduce, if we still rely on such a low targetSdk.

**Comment by TPS** [comment]
:+1: +1 for #504 resolution, but then we ***must*** make actual progress on this _specific_ issue!  Anyone?!

**Comment by SecUpwN** [comment]
>...we must make actual progress on this specific issue! Anyone?!

@TPS, please don't push us. We still "feature" more bugs than functionality.

>As far as I know its a best practice to keep
>- `targetSdkVersion` and `compileSdkVersion` as high as possible

Ok, what do you suggest to which versions to raise them to?

>  - in order to be able to use newer features on newer phones
>- `minSdkVersion` as low as possible
>  - in order to be able to run on as many devices as possible
>  - @SecUpwN this value decides our compability for older androids

Fully agree. Again, which version do you suggest here?

**Comment by Klaws--** [comment]
I'd go for numbers. NOT roman numerals if these can pose ambiguities for a dual-SIM setup - otherwise okay.

I totally agree that numbers do not easily convey the precise meaning of a threat level. The same is however true for colors, bar graphs or whatever else you can fit into a 5mm status icon. I mean, was yellow/2/two sparks/half-gray-tower more like "Insecure Service Area: At least 1 new or unknown BTS is in operation in this vicinity, or using an encryption < A5/3." or more like "Hostile Service Actions Detected: An IMSI-Catcher is actively tracking users." (first answer would be correct). And does that mean I am in danger, or in danger when I use my phone, or only when I perform potentially suspicious communication over my phone?

The status icon should just give a clue that something is going on, with the numeric (or whatever else) indicator giving an additional quick hint for the power user. The causal user will need to have closer look at AIMSICD to receive a human-readable and -understandable summary of what is going on and, perhaps, how it affects him or others in the vicinity (IMSI catchers are used in find-fix-finish operations - SOPs suggest to employ massive firepower during the "fix" stage already, and one does not want to turn from an "innocent bystander" into, um, "collateral damage").

**Comment by Klaws--** [comment]
Also wanted to mention the DEFCON levels (1-5), because everyone knows them. Well, at least everyone who has watched "War Games" in the 80s.

Unfortunately, the  DEFCON level also comes with a color scheme. blue (5), green (4), yellow (3), red (2), white (1). Google wants white icons, so it would mean "Nuclear War Imminent" all the time.

Yuk.

**Comment by SecUpwN** [comment]
@larsgrefer, since you've implemented #818, do you think we can close this, or shall we continue to discuss this?

**Comment by redbassett** [comment]
#818 is almost exactly what I envisioned when I opened this issue. I'm all for the monochrome status bar icon and full-colour notification. My Phone happens to be from Republic Wireless, and this is exactly what they do with their persistent notification for system status.

**Comment by TPS** [comment]
_(@redbassett If you're on Republic Wireless right now, your backend is Sprint CDMA, like me on Ting, so #489 & #457 are particularly relevant to you, also.  It'd be great if you chime in, there, too.)_
